### ISSUE TITLE
Could not load dynamic library 'libcudart.so.10.1'

### ISSUE THREAD

**Issue by asfarley** [tensorflow/tensorboard#4504]
I'm following a tutorial from Ultralytics which uses tensorboard for plotting. They directed me here for this issue.
[https://github.com/ultralytics/yolov3/issues/1617](https://github.com/ultralytics/yolov3/issues/1617)

When I run the training routine, I get the following error message from tensorboard:

```
Start Tensorboard with "tensorboard --logdir runs/train", view at http://localhost:6006/
2020-12-22 00:02:45.814933: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcudart.so.10.1'; dlerror: libcudart.so.10.1: cannot open shared object file: No such file or directory
2020-12-22 00:02:45.814953: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
```
I've searched StackOverflow which has a similar question posted:
[https://stackoverflow.com/questions/64880546/tensorflow-could-not-load-dynamic-library-libcudart-so-10-0-on-ubuntu-18-04](https://stackoverflow.com/questions/64880546/tensorflow-could-not-load-dynamic-library-libcudart-so-10-0-on-ubuntu-18-04)

However, the above case seems to be slightly different from my situation, and in my experience Stack Overflow dislikes issues related to e.g. version compatibility.

Here's my output from diagnose_tensorboard.py:
### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
INFO: diagnose_tensorboard.py version ad749c17953187cae68d6c455e8b3fabd3e69096

--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=8, micro=3, releaselevel='final', serial=0)
INFO: os.name: posix
INFO: os.uname(): posix.uname_result(sysname='Linux', nodename='alexander-MS-7B16', release='5.4.0-58-generic', version='#64-Ubuntu SMP Wed Dec 9 08:16:25 UTC 2020', machine='x86_64')
INFO: sys.getwindowsversion(): N/A

--- check: package_management
INFO: has conda-meta: True
INFO: $VIRTUAL_ENV: None

--- check: installed_packages
INFO: installed: tensorboard==2.4.0
INFO: installed: tensorflow==2.3.1
INFO: installed: tensorflow-estimator==2.3.0

--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.4.0'

--- check: tensorflow_python_version
2020-12-22 00:06:47.852905: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcudart.so.10.1'; dlerror: libcudart.so.10.1: cannot open shared object file: No such file or directory
2020-12-22 00:06:47.852923: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO: tensorflow.__version__: '2.3.1'
INFO: tensorflow.__git_version__: 'v2.3.0-54-gfcc4b966f1'

--- check: tensorboard_binary_path
INFO: which tensorboard: b'/home/alexander/anaconda3/bin/tensorboard\n'

--- check: addrinfos
socket.has_ipv6 = True
socket.AF_UNSPEC = <AddressFamily.AF_UNSPEC: 0>
socket.SOCK_STREAM = <SocketKind.SOCK_STREAM: 1>
socket.AI_ADDRCONFIG = <AddressInfo.AI_ADDRCONFIG: 32>
socket.AI_PASSIVE = <AddressInfo.AI_PASSIVE: 1>
Loopback flags: <AddressInfo.AI_ADDRCONFIG: 32>
Loopback infos: [(<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::1', 0, 0, 0)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 0))]
Wildcard flags: <AddressInfo.AI_PASSIVE: 1>
Wildcard infos: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('0.0.0.0', 0)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::', 0, 0, 0))]

--- check: readable_fqdn
INFO: socket.getfqdn(): 'alexander-MS-7B16'

--- check: stat_tensorboardinfo
INFO: directory: /tmp/.tensorboard-info
INFO: .tensorboard-info directory does not exist

--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['/home/alexander/anaconda3/lib/python3.8/site-packages']; bad_roots (0): []

--- check: full_pip_freeze
INFO: pip freeze --all:
absl-py==0.11.0
alabaster==0.7.12
anaconda-client==1.7.2
anaconda-navigator==1.9.12
anaconda-project==0.8.3
argh==0.26.2
asn1crypto==1.3.0
astroid @ file:///tmp/build/80754af9/astroid_1592495912941/work
astropy==4.0.1.post1
astunparse==1.6.3
atomicwrites==1.4.0
attrs==19.3.0
autopep8 @ file:///tmp/build/80754af9/autopep8_1592412889138/work
Babel==2.8.0
backcall==0.2.0
backports.functools-lru-cache==1.6.1
backports.shutil-get-terminal-size==1.0.0
backports.tempfile==1.0
backports.weakref==1.0.post1
beautifulsoup4==4.9.1
bitarray @ file:///tmp/build/80754af9/bitarray_1594753876414/work
bkcharts==0.2
bleach==3.1.5
bokeh @ file:///tmp/build/80754af9/bokeh_1593187602660/work
boto==2.49.0
Bottleneck==1.3.2
brotlipy==0.7.0
cachetools==4.1.1
certifi==2020.6.20
cffi==1.14.0
chardet==3.0.4
click==7.1.2
cloudpickle @ file:///tmp/build/80754af9/cloudpickle_1594141588948/work
clyent==1.2.2
colorama==0.4.3
conda==4.8.3
conda-build==3.18.11
conda-package-handling==1.7.0
conda-verify==3.4.2
contextlib2==0.6.0.post1
cryptography==2.9.2
cycler==0.10.0
Cython @ file:///tmp/build/80754af9/cython_1594831566883/work
cytoolz==0.10.1
dask @ file:///tmp/build/80754af9/dask-core_1594156306305/work
dataclasses==0.6
decorator==4.4.2
defusedxml==0.6.0
diff-match-patch @ file:///tmp/build/80754af9/diff-match-patch_1594828741838/work
distributed @ file:///tmp/build/80754af9/distributed_1594750293818/work
docutils==0.16
entrypoints==0.3
et-xmlfile==1.0.1
fastcache==1.1.0
filelock==3.0.12
flake8==3.8.3
Flask==1.1.2
fsspec==0.7.4
future==0.18.2
gast==0.3.3
gevent @ file:///tmp/build/80754af9/gevent_1593009537125/work
glob2==0.7
gmpy2==2.0.8
google-auth==1.23.0
google-auth-oauthlib==0.4.2
google-pasta==0.2.0
greenlet==0.4.16
grpcio==1.33.2
h5py==2.10.0
HeapDict==1.0.1
html5lib @ file:///tmp/build/80754af9/html5lib_1593446221756/work
idna @ file:///tmp/build/80754af9/idna_1593446292537/work
imageio @ file:///tmp/build/80754af9/imageio_1594161405741/work
imagesize==1.2.0
importlib-metadata @ file:///tmp/build/80754af9/importlib-metadata_1593446406207/work
intervaltree @ file:///tmp/build/80754af9/intervaltree_1594361675072/work
ipykernel @ file:///tmp/build/80754af9/ipykernel_1594753368510/work/dist/ipykernel-5.3.2-py3-none-any.whl
ipython @ file:///tmp/build/80754af9/ipython_1593447377063/work
ipython-genutils==0.2.0
ipywidgets==7.5.1
isort==4.3.21
itsdangerous==1.1.0
jdcal==1.4.1
jedi @ file:///tmp/build/80754af9/jedi_1592841866100/work
jeepney==0.4.3
Jinja2==2.11.2
joblib @ file:///tmp/build/80754af9/joblib_1594236160679/work
json5==0.9.5
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client @ file:///tmp/build/80754af9/jupyter_client_1594826976318/work
jupyter-console==6.1.0
jupyter-core==4.6.3
jupyterlab==2.1.5
jupyterlab-server @ file:///tmp/build/80754af9/jupyterlab_server_1594164409481/work
Keras-Preprocessing==1.1.2
keyring @ file:///tmp/build/80754af9/keyring_1593109721646/work
kiwisolver==1.2.0
lazy-object-proxy==1.4.3
libarchive-c==2.9
llvmlite==0.33.0+1.g022ab0f
locket==0.2.0
lxml @ file:///tmp/build/80754af9/lxml_1594826779415/work
Markdown==3.3.3
MarkupSafe==1.1.1
matplotlib @ file:///tmp/build/80754af9/matplotlib-base_1592846008246/work
mccabe==0.6.1
mistune==0.8.4
mkl-fft==1.1.0
mkl-random==1.1.1
mkl-service==2.3.0
mock==4.0.2
more-itertools==8.4.0
mpmath==1.1.0
msgpack==1.0.0
multipledispatch==0.6.0
navigator-updater==0.2.1
nbconvert==5.6.1
nbformat==5.0.7
networkx @ file:///tmp/build/80754af9/networkx_1594377231366/work
nltk @ file:///tmp/build/80754af9/nltk_1592496090529/work
nose==1.3.7
notebook==6.0.3
numba==0.50.1
numexpr==2.7.1
numpy==1.18.5
numpydoc @ file:///tmp/build/80754af9/numpydoc_1594166760263/work
oauthlib==3.1.0
olefile==0.46
opencv-python==4.4.0.46
openpyxl @ file:///tmp/build/80754af9/openpyxl_1594167385094/work
opt-einsum==3.3.0
packaging==20.4
pandas @ file:///tmp/build/80754af9/pandas_1592841659173/work
pandocfilters==1.4.2
parso==0.7.0
partd==1.1.0
path==13.1.0
pathlib2==2.3.5
pathtools==0.1.2
patsy==0.5.1
pep8==1.7.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow @ file:///tmp/build/80754af9/pillow_1594307295532/work
pip==20.1.1
pkginfo==1.5.0.1
pluggy==0.13.1
ply==3.11
prometheus-client==0.8.0
prompt-toolkit==3.0.5
protobuf==3.14.0
psutil==5.7.0
ptyprocess==0.6.0
py @ file:///tmp/build/80754af9/py_1593446248552/work
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycocotools==2.0.2
pycodestyle==2.6.0
pycosat==0.6.3
pycparser @ file:///tmp/build/80754af9/pycparser_1594388511720/work
pycurl==7.43.0.5
pydocstyle @ file:///tmp/build/80754af9/pydocstyle_1592848020240/work
pyflakes==2.2.0
Pygments==2.6.1
pylint @ file:///tmp/build/80754af9/pylint_1592496053748/work
pyodbc===4.0.0-unsupported
pyOpenSSL @ file:///tmp/build/80754af9/pyopenssl_1594392929924/work
pyparsing==2.4.7
pyrsistent==0.16.0
PySocks==1.7.1
pytest==5.4.3
python-dateutil==2.8.1
python-jsonrpc-server @ file:///tmp/build/80754af9/python-jsonrpc-server_1594397536060/work
python-language-server @ file:///tmp/build/80754af9/python-language-server_1594161909948/work
pytz==2020.1
PyWavelets==1.1.1
pyxdg==0.26
PyYAML==5.3.1
pyzmq==19.0.1
QDarkStyle==2.8.1
QtAwesome==0.7.2
qtconsole @ file:///tmp/build/80754af9/qtconsole_1592848611704/work
QtPy==1.9.0
regex @ file:///tmp/build/80754af9/regex_1593435547652/work
requests @ file:///tmp/build/80754af9/requests_1592841827918/work
requests-oauthlib==1.3.0
rope==0.17.0
rsa==4.6
Rtree==0.9.4
ruamel-yaml==0.15.87
scikit-image==0.16.2
scikit-learn @ file:///tmp/build/80754af9/scikit-learn_1592502866053/work
scipy @ file:///tmp/build/80754af9/scipy_1592930511789/work
seaborn==0.10.1
SecretStorage==3.1.2
Send2Trash==1.5.0
setuptools==49.2.0.post20200714
simplegeneric==0.8.1
singledispatch==3.4.0.3
sip==4.19.13
six==1.15.0
snowballstemmer==2.0.0
sortedcollections==1.2.1
sortedcontainers==2.2.2
soupsieve==2.0.1
Sphinx @ file:///tmp/build/80754af9/sphinx_1594223420021/work
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==1.0.3
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.4
sphinxcontrib-websupport @ file:///tmp/build/80754af9/sphinxcontrib-websupport_1593446360927/work
spyder @ file:///tmp/build/80754af9/spyder_1594832131358/work
spyder-kernels @ file:///tmp/build/80754af9/spyder-kernels_1594751551963/work
SQLAlchemy @ file:///tmp/build/80754af9/sqlalchemy_1593446299926/work
statsmodels==0.11.1
sympy @ file:///tmp/build/80754af9/sympy_1594236545715/work
tables==3.6.1
tblib==1.6.0
tensorboard==2.4.0
tensorboard-plugin-wit==1.7.0
tensorflow==2.3.1
tensorflow-estimator==2.3.0
termcolor==1.1.0
terminado==0.8.3
terminaltables==3.1.0
testpath==0.4.4
thop==0.0.31.post2005241907
threadpoolctl @ file:///tmp/tmp9twdgx9k/threadpoolctl-2.1.0-py3-none-any.whl
toml @ file:///tmp/build/80754af9/toml_1592853716807/work
toolz==0.10.0
torch==1.7.0
torchvision==0.8.1
tornado==6.0.4
tqdm @ file:///tmp/build/80754af9/tqdm_1593446365756/work
traitlets==4.3.3
typing-extensions @ file:///tmp/build/80754af9/typing_extensions_1592847887441/work
ujson==1.35
unicodecsv==0.14.1
urllib3==1.25.9
watchdog @ file:///tmp/build/80754af9/watchdog_1593447344699/work
wcwidth @ file:///tmp/build/80754af9/wcwidth_1593447189090/work
webencodings==0.5.1
Werkzeug==1.0.1
wheel==0.34.2
widgetsnbextension==3.5.1
wrapt==1.11.2
wurlitzer @ file:///tmp/build/80754af9/wurlitzer_1594753850195/work
xlrd==1.2.0
XlsxWriter==1.2.9
xlwt==1.3.0
xmltodict==0.12.0
yapf @ file:///tmp/build/80754af9/yapf_1593528177422/work
zict==2.0.0
zipp==3.1.0
zope.event==4.4
zope.interface==4.7.1

``````

</details>

### Next steps

No action items identified. Please copy ALL of the above output,
including the lines containing only backticks, into your GitHub issue
or comment. Be sure to redact any sensitive information.

The output from `nvidia-smi` is:
```
Tue Dec 22 00:12:01 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.80.02    Driver Version: 450.80.02    CUDA Version: 11.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce GTX 1070    Off  | 00000000:01:00.0  On |                  N/A |
|  0%   36C    P8    15W / 200W |    262MiB /  8116MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A       829      G   /usr/lib/xorg/Xorg                 35MiB |
|    0   N/A  N/A      1599      G   /usr/lib/xorg/Xorg                129MiB |
|    0   N/A  N/A      1725      G   /usr/bin/gnome-shell               80MiB |
|    0   N/A  N/A     11473      G   /usr/lib/firefox/firefox            2MiB |
+-----------------------------------------------------------------------------+
```

My steps for creating this issue are:
1) Running Ubuntu 20.04
2) Install CUDA 11.0
3) Install PyTorch
4) Install all packages for Ultralytics tutorial via `pip -r requirements.txt`
5) Test running Yolo with Ultralytics PyTorch script; runs correctly, appears to find and use CUDA
6) Test training a Yolo network with Ultralytics PyTorch script; get error above

Basically, my question is:  Does tensorboard require CUDA 10.1? Is it incompatible with 11.0?

**Comment by arghyaganguly** [comment]
Could you please try downgrading to CUDA 10.1 and check if the issue still exists?

**Comment by glenn-jocher** [comment]
@asfarley @arghyaganguly incidentally this issue is not quite reproducible in the YOLOv5 colab notebook. Instead there we get a different message:

```bash
Using torch 1.7.0+cu101 CUDA:0 (Tesla V100-SXM2-16GB, 16130MB)
...
Start Tensorboard with "tensorboard --logdir runs/train", view at http://localhost:6006/
2020-12-22 21:21:01.875591: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1
```

nvidia-smi output there shows  
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.27.04    Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:04.0 Off |                    0 |
| N/A   37C    P0    43W / 300W |     10MiB / 16130MiB |      0%      Default |
|                               |                      |                 ERR! |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

**Comment by asfarley** [comment]
After downgrading to CUDA 10.1, the error message goes away. I don't really know why I had 11.0 installed anyway, I guess just to have the latest, but it seems like that might have been a bad choice.
